core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/DiagramServices.java

```diff
@@ -680,6 +681,43 @@ public boolean isOnDiagram(DDiagram diagramTarget, EObject semanticTarget) {
         || ViewpointPackage.Literals.DREPRESENTATION_ELEMENT__SEMANTIC_ELEMENTS.equals(setting.getEStructuralFeature());
   }
 
+  /**
+   * Return whether the given node mapping can be created into the given container 
+   */
+  public boolean isValidMapping(DSemanticDiagram diagram, AbstractNodeMapping mapping_p,
+      DSemanticDecorator container) {
+
```

Comment:
It might be interesting to take a look at `org.eclipse.sirius.diagram.business.internal.query.AbstractNodeMappingApplicabilityTester` class. 
It has methods such as `canCreateIn(DDiagram diagram)` , `canCreateIn(DDiagramElementContainer container)` ,`canCreateIn(DNode node)`. Maybe we can delegate some (or all) of this work to this class.